text-motions.c

```diff
@@ -586,25 +586,37 @@ size_t text_function_end_prev(Text *txt, size_t pos) {
 	return text_function_end_direction(txt, pos, -1);
 }
 
-size_t text_block_start(Text *txt, size_t pos) {
+static size_t text_block_direction(Text *txt, size_t pos, int direction) {
 	int nest = 0;
-	char c;
+	char c, target, nest_target;
 	bool instring = false;
 	Iterator it = text_iterator_get(txt, pos);
-	while (text_iterator_byte_prev(&it, &c)) {
+
+	target = (direction > 0 ? '}' : '{');
+	nest_target = (direction > 0 ? '{' : '}');
```

Comment:
Considering that `vis` is not trying to be compatible with Vim’s bugs and limitations, does it really make send
sense to add hard wired motions that only apply to languages with a C-like syntax? I think this kind of, for lack of a better word, semantic motions would be better if they tied in with the lexing system, which would also deliver non-naive handling of blocks to be ignored (comments, here doc, strings with all kinds of quotes, interpolation, misc. and sundry).